### PR TITLE
Added polycos argument to photonphase

### PR DESCRIPTION
**Pull request by sashalvna**
**Fixed version of feature-photonphase-polycos which only includes the necessary changes.
A polycos argument has been added to photonphase.py, which uses polycos to calculate pulse phases without using the full model of the pulsar. Useful for large files, where photonphase.py previously used to crash/time out. A test of the polycos argument has also been added to test_photonphase.py.

### CODE REVIEW

File: tests/test_photonphase.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 import pytest
 
-import pint.scripts.photonphase as photonphase
+import photonphase
```
Comment (paulray) [inline tests/test_photonphase.py:7]: I'm wondering why this change was needed?

Comment (paulray) [reply tests/test_photonphase.py:7]: Actually, it seems to be causing test failures. I think this needs to be reverted.

**Comment by paulray** [comment]
Thanks so much for doing this!  I'll close the other one now.

File: src/pint/scripts/photonphase.py [hunk 2]
```diff
@@ -15,9 +15,7 @@
 from pint.observatory.satellite_obs import get_satellite_observatory
 from pint.plot_utils import phaseogram_binned
 from pint.pulsar_mjd import Time
-
-log = logging.getLogger(__name__)
-log.setLevel("INFO")
+import polycos
```
Comment (paulray) [inline src/pint/scripts/photonphase.py:18]: This should be `import pint.polycos` I believe

**Comment by sashalvna** [comment]
> 

Sounds good, I will do that. Do I just add it to the most recent section (0.8.5), under "Added"?

**Comment by paulray** [comment]
> > 
> 
> Sounds good, I will do that. Do I just add it to the most recent section (0.8.5), under "Added"?

There should be an [Unreleased] section above the 0.8.5 one. We collect all the changes there and then when we do a new release (e.g. 0.8.6), I rename Unreleased to 0.8.6.

**Comment by sashalvna** [comment]
I don't see that section when I open CHANGELOG.md, but I believe that the branch should have been up to date. Is there something I need to do to be able to see it?

**Comment by paulray** [comment]
Sorry, I think it just got added a few hours ago. Can you merge nanograv/master into your branch?

**Comment by paulray** [comment]
Or if that is a pain, don't worry. It is trivial for me to add that line manually after merging.

**Comment by sashalvna** [comment]
I was able to merge it into my branch and update the changelog, and fixed imports in photonphase.py.

**Comment by paulray** [comment]
Excellent! Did you see the other comments I posted?

**Comment by sashalvna** [comment]
I saw the change of "import polycos" to "import pint.polycos". Were there any others I missed? I'll double check.

**Review by paulray — COMMENTED** [review]
OK, I made a few little changes and resolved some conflicts that got in while this PR was pending. It looks great! The only comment remaining is whether we should add a high precision test of polycos vs the normal method for a couple different examples.

**Comment by paulray** [comment]
Ah, we also need to run `black` on photonphase.py and test_photonphase.py

**Comment by paulray** [comment]
Oh, and note that I committed directly to this PR. Not sure exactly how that works, but make sure to `git pull` before making any more edits on your side.

File: tests/test_photonphase.py [hunk 3]
```diff
@@ -127,6 +127,20 @@ def test_OrbPhase_column():
...
+    "Check that barycentered NICER data is processed correctly with --polyco."
+    cmd = "--polycos {0} {1}".format(eventfile_nicer, parfile_nicer)
+    photonphase.main(cmd.split())
+    out, err = capsys.readouterr()
+    v = 0.0
+    for l in out.split("\n"):
+        if l.startswith("Htest"):
+            v = float(l.split()[2])
+    # Check that H-test is 216.67
+    assert abs(v - 216.67) < 1
```
[the remark this answers is not in the document]
Comment (sashalvna) [reply tests/test_photonphase.py:143]: I can try and write a test that does that. Should I compare the first few phase values it calculates?

**Comment by sashalvna** [comment]
> Ah, we also need to run `black` on photonphase.py and test_photonphase.py

How would I do that? I'm not certain what that is, but it seems that is the only test that is failing, right?

**Comment by paulray** [comment]
If you have black installed just  run like this: `black photonphase.py`. It is installed automatically if you `pip install -r requirements_dev.txt` which will make sure you have everything installed that is needed for development and testing of the code.

File: tests/test_photonphase.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (paulray) [reply tests/test_photonphase.py:143]: Since numpy easily operates on vectors, I'd probably set some limit on `(A-B).std()` (if A and B are the arrays of phases computed with the two methods) so you are testing all the values not just a few.

**Comment by sashalvna** [comment]
Since I did that git pull yesterday, I have been having issues running scripts (or the `pip install -r requirements_dev.txt`) because logging.py has been added to PINT/src/pint, and now most scripts are unable to find the correct modules in the imports.

**Comment by paulray** [comment]
> Since I did that git pull yesterday, I have been having issues running scripts (or the `pip install -r requirements_dev.txt`) because logging.py has been added to PINT/src/pint, and now most scripts are unable to find the correct modules in the imports.

Hmm. Can you post the error messages you are getting? Also, what kind of python install are you using (conda, virtualenv, ?) and how did you install pint (e.g. `pip install -e .`)?

**Comment by paulray** [comment]
If it is just loguru, then just `pip install loguru`

**Comment by sashalvna** [comment]
> If it is just loguru, then just `pip install loguru`

I got this error message when I tried it:
image

**Comment by paulray** [comment]
Hmm, that should not be happening. Are you currently in `src/pint`? If so move up to the top level PINT source dir.
It seems like pip is doing an `import logging` and getting `src/pint/logging.py` instead of the system logging.py.
Make sure your PYTHONPATH does NOT have `src /pint` in it, and maybe try running `python` then `import logging` and then `print(logging.__file__)`

**Comment by sashalvna** [comment]
> Hmm, that should not be happening. Are you currently in `src/pint`? If so move up to the top level PINT source dir. It seems like pip is doing an `import logging` and getting `src/pint/logging.py` instead of the system logging.py. Make sure your PYTHONPATH does NOT have `src /pint` in it, and maybe try running `python` then `import logging` and then `print(logging.__file__)`

I fixed it, it was an issue with PYTHONPATH. Now everything works and I was able to run black on both photonphase.py and test_photonphase.py

**Comment by paulray** [comment]
Yay!

File: tests/test_photonphase.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sashalvna) [reply tests/test_photonphase.py:147]: I have added the test, but am not certain what to set the limit to. It passes the test up to 1e-8.

[the remark this answers is not in the document]
Comment (paulray) [reply tests/test_photonphase.py:147]: That is fantastic!
